spark-etl/src/main/scala/geotrellis/spark/etl/Etl.scala

```diff
@@ -24,27 +24,25 @@ object Etl {
 
   def ingest[
     I: ProjectedExtentComponent: TypeTag: ? => TilerKeyMethods[I, K],
-    K: SpatialComponent: TypeTag,
-    V <: CellGrid: TypeTag: Stitcher: ? => TileReprojectMethods[V]: ? => CropMethods[V]: ? => TileMergeMethods[V]: ? => TilePrototypeMethods[V]
-  ](args: Seq[String], keyIndexMethod: KeyIndexMethod[K], modules: Seq[TypedModule] = Etl.defaultModules)(implicit sc: SparkContext) = {
+    K: SpatialComponent: Boundable: TypeTag,
+    V <: CellGrid: TypeTag: Stitcher: (? => TileReprojectMethods[V]): (? => CropMethods[V]): (? => TileMergeMethods[V]): (? => TilePrototypeMethods[V])
+  ](
+    args: Seq[String], keyIndexMethod: KeyIndexMethod[K], modules: Seq[TypedModule] = Etl.defaultModules
+  )(implicit sc: SparkContext) = {
     implicit def classTagK = ClassTag(typeTag[K].mirror.runtimeClass(typeTag[K].tpe)).asInstanceOf[ClassTag[K]]
     implicit def classTagV = ClassTag(typeTag[V].mirror.runtimeClass(typeTag[V].tpe)).asInstanceOf[ClassTag[V]]
 
     val etl = Etl(args)
     val sourceTiles = etl.load[I, V]
-    val (_, metadata) = etl.collectMetadata(sourceTiles)
-    val tiled = sourceTiles.cutTiles[K](metadata, NearestNeighbor)
-    val (zoom, reprojected) = etl.reproject(ContextRDD(tiled, metadata))
-    etl.save[K, V](LayerId(etl.conf.layerName(), zoom), reprojected, keyIndexMethod)
+    val (tiledZoom, tiled) = etl.tile(sourceTiles)
```

Comment:
Probably that's what ew really wanted, awesome that it derives type `K` here. Think that we really don't need to get *only* metadata with etl object, but probably that could be useful. 
Anyway your approach looks nicer think :+1: